### ISSUE TITLE
javax.servlet.ServletException: Resource class org.apache.oodt.pcs.services.HealthResource can not be instantiated due to InvocationTargetException

### ISSUE THREAD

**Issue by lewismc** [apache/drat#41]
When running DRAT master branch on the recent [Apache OODT 0.10 RC#1](http://www.mail-archive.com/dev%40oodt.apache.org/msg05909.html) I am coming across some problems running drat go.
A stack trace can be located within logs/cas_resource1.log

```
SEVERE: Servlet /pcs threw load() exception
javax.servlet.ServletException: Resource class org.apache.oodt.pcs.services.HealthResource can not be instantiated due to InvocationTargetException
    at org.apache.cxf.jaxrs.servlet.CXFNonSpringJaxrsServlet.createSingletonInstance(CXFNonSpringJaxrsServlet.java:247)
    at org.apache.cxf.jaxrs.servlet.CXFNonSpringJaxrsServlet.getResourceProviders(CXFNonSpringJaxrsServlet.java:211)
    at org.apache.cxf.jaxrs.servlet.CXFNonSpringJaxrsServlet.loadBus(CXFNonSpringJaxrsServlet.java:92)
    at org.apache.cxf.transport.servlet.AbstractCXFServlet.init(AbstractCXFServlet.java:84)
    at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1139)
    at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:966)
    at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:3956)
    at org.apache.catalina.core.StandardContext.start(StandardContext.java:4230)
    at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:760)
    at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:740)
    at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:544)
    at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:920)
    at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:883)
    at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:492)
    at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1138)
    at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:311)
    at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:120)
    at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1022)
    at org.apache.catalina.core.StandardHost.start(StandardHost.java:736)
    at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1014)
    at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:443)
    at org.apache.catalina.core.StandardService.start(StandardService.java:448)
    at org.apache.catalina.core.StandardServer.start(StandardServer.java:700)
    at org.apache.catalina.startup.Catalina.start(Catalina.java:552)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:295)
    at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:433)

Aug 31, 2015 9:26:31 PM org.apache.jasper.compiler.JspRuntimeContext <init>
FINE: Parent class loader is: WebappClassLoader
  delegate: false
  repositories:
    /WEB-INF/classes/
----------> Parent Classloader:
org.apache.catalina.loader.StandardClassLoader@6220ac02
```

... investigating

**Comment by lewismc** [comment]
drat.log exposes the following

```
INFO: Executing task: [MimePartitioner] locally
Aug 31, 2015 9:30:11 PM org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder buildImports
INFO: Loading PgeConfig file '/usr/local/drat/deploy//pge/policy/PgeConfig_MimePartitioner.xml'
Aug 31, 2015 9:30:11 PM org.apache.oodt.cas.pge.PGETaskInstance createExeDir
INFO: Creating PGE execution working directory: [/usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530]
Aug 31, 2015 9:30:11 PM org.apache.oodt.cas.pge.PGETaskInstance updateStatus
INFO: Updating status to workflow as BUILDING CONFIG FILE
Aug 31, 2015 9:30:11 PM org.apache.oodt.cas.pge.PGETaskInstance updateStatus
INFO: Updating status to workflow as PGE EXEC
Aug 31, 2015 9:30:11 PM org.apache.oodt.commons.io.LoggerOutputStream flush
INFO: OUTPUT: Creating working dirs
OUTPUT: Running MIME partitioner

Aug 31, 2015 9:30:11 PM org.apache.oodt.commons.io.LoggerOutputStream flush
INFO: OUTPUT: Creating working dirs
OUTPUT: Running MIME partitioner

Aug 31, 2015 9:30:11 PM org.apache.oodt.commons.io.LoggerOutputStream flush
SEVERE: ERROR:   File "/usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py", line 111
ERROR: SyntaxError: Non-ASCII character '\xe2' in file /usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py on line 111, but no encoding declared; see http://www.python.org/peps/pep-0263.html for details

Aug 31, 2015 9:30:11 PM org.apache.oodt.commons.io.LoggerOutputStream flush
SEVERE: ERROR:   File "/usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py", line 111
ERROR: SyntaxError: Non-ASCII character '\xe2' in file /usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py on line 111, but no encoding declared; see http://www.python.org/peps/pep-0263.html for details

java.lang.RuntimeException: Pge didn't finish successfully
    at org.apache.oodt.cas.pge.PGETaskInstance.runPge(PGETaskInstance.java:322)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:451)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
org.apache.oodt.cas.workflow.structs.exceptions.WorkflowTaskInstanceException: PGETask failed : Exception when executing PGE commands '[export PATH=$HOME/bin/:${PATH}, shopt -s expand_aliases, echo "Creating working dirs", mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/input ; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/output; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs, echo "Running MIME partitioner", /usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py -u http://localhost:8080/solr/drat -c 100 -w http://localhost:9001 -t urn:drat:RatCodeAudit > /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs/mime-partitioner-1441081811530.log]' : Pge didn't finish successfully
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:456)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.Exception: Exception when executing PGE commands '[export PATH=$HOME/bin/:${PATH}, shopt -s expand_aliases, echo "Creating working dirs", mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/input ; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/output; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs, echo "Running MIME partitioner", /usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py -u http://localhost:8080/solr/drat -c 100 -w http://localhost:9001 -t urn:drat:RatCodeAudit > /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs/mime-partitioner-1441081811530.log]' : Pge didn't finish successfully
    at org.apache.oodt.cas.pge.PGETaskInstance.runPge(PGETaskInstance.java:334)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:451)
    ... 4 more
Aug 31, 2015 9:30:11 PM org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread executeTaskLocally
WARNING: Exception executing task: [MimePartitioner] locally: Message: PGETask failed : Exception when executing PGE commands '[export PATH=$HOME/bin/:${PATH}, shopt -s expand_aliases, echo "Creating working dirs", mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/input ; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/output; mkdir /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs, echo "Running MIME partitioner", /usr/local/drat/deploy//pge/bin/mime_partitioner/mime_rat_partitioner.py -u http://localhost:8080/solr/drat -c 100 -w http://localhost:9001 -t urn:drat:RatCodeAudit > /usr/local/drat/deploy//data/jobs/mimepartitioner/1441081811530/logs/mime-partitioner-1441081811530.log]' : Pge didn't finish successfully
Aug 31, 2015 9:30:17 PM org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread checkTaskRequiredMetadata
INFO: Task: [RatAggregator] has no required metadata fields
Aug 31, 2015 9:30:17 PM org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread executeTaskLocally
INFO: Executing task: [RatAggregator] locally
Aug 31, 2015 9:30:17 PM org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder buildImports
INFO: Loading PgeConfig file '/usr/local/drat/deploy//pge/policy/PgeConfig_RatAggregator.xml'
Aug 31, 2015 9:30:17 PM org.apache.oodt.cas.filemgr.catalog.LuceneCatalog paginateQuery
WARNING: Query: [q=] for Product Type: [urn:drat:RatLog] returned no results
java.lang.NullPointerException
    at org.apache.oodt.cas.filemgr.system.XmlRpcFileManager.complexQuery(XmlRpcFileManager.java:607)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.apache.xmlrpc.Invoker.execute(Invoker.java:130)
    at org.apache.xmlrpc.XmlRpcWorker.invokeHandler(XmlRpcWorker.java:84)
    at org.apache.xmlrpc.XmlRpcWorker.execute(XmlRpcWorker.java:146)
    at org.apache.xmlrpc.XmlRpcServer.execute(XmlRpcServer.java:139)
    at org.apache.xmlrpc.XmlRpcServer.execute(XmlRpcServer.java:125)
    at org.apache.xmlrpc.WebServer$Connection.run(WebServer.java:761)
    at org.apache.xmlrpc.WebServer$Runner.run(WebServer.java:642)
    at java.lang.Thread.run(Thread.java:745)
org.apache.xmlrpc.XmlRpcException: java.lang.Exception: org.apache.oodt.cas.filemgr.structs.exceptions.CatalogException: Failed to perform complex query : null
    at org.apache.xmlrpc.XmlRpcClientResponseProcessor.decodeException(XmlRpcClientResponseProcessor.java:104)
    at org.apache.xmlrpc.XmlRpcClientResponseProcessor.decodeResponse(XmlRpcClientResponseProcessor.java:71)
    at org.apache.xmlrpc.XmlRpcClientWorker.execute(XmlRpcClientWorker.java:73)
    at org.apache.xmlrpc.XmlRpcClient.execute(XmlRpcClient.java:194)
    at org.apache.xmlrpc.XmlRpcClient.execute(XmlRpcClient.java:185)
    at org.apache.xmlrpc.XmlRpcClient.execute(XmlRpcClient.java:178)
    at org.apache.oodt.cas.filemgr.system.XmlRpcFileManagerClient.complexQuery(XmlRpcFileManagerClient.java:1003)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.fillIn(XmlFilePgeConfigBuilder.java:322)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.fillIn(XmlFilePgeConfigBuilder.java:313)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.getCustomMetadata(XmlFilePgeConfigBuilder.java:180)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.build(XmlFilePgeConfigBuilder.java:109)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.buildImports(XmlFilePgeConfigBuilder.java:101)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.build(XmlFilePgeConfigBuilder.java:69)
    at org.apache.oodt.cas.pge.PGETaskInstance.createPgeConfig(PGETaskInstance.java:133)
    at org.apache.oodt.cas.pge.PGETaskInstance.initialize(PGETaskInstance.java:98)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:449)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
java.lang.Exception: Failed to parse value: SQL(FORMAT='$FileLocation/$Filename',SORT_BY='CAS.ProductReceivedTime'){SELECT FileLocation,Filename,CAS.ProductReceivedTime FROM RatLog}
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.fillIn(XmlFilePgeConfigBuilder.java:326)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.fillIn(XmlFilePgeConfigBuilder.java:313)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.getCustomMetadata(XmlFilePgeConfigBuilder.java:180)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.build(XmlFilePgeConfigBuilder.java:109)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.buildImports(XmlFilePgeConfigBuilder.java:101)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.build(XmlFilePgeConfigBuilder.java:69)
    at org.apache.oodt.cas.pge.PGETaskInstance.createPgeConfig(PGETaskInstance.java:133)
    at org.apache.oodt.cas.pge.PGETaskInstance.initialize(PGETaskInstance.java:98)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:449)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.oodt.cas.filemgr.structs.exceptions.CatalogException: java.lang.Exception: org.apache.oodt.cas.filemgr.structs.exceptions.CatalogException: Failed to perform complex query : null
    at org.apache.oodt.cas.filemgr.system.XmlRpcFileManagerClient.complexQuery(XmlRpcFileManagerClient.java:1009)
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.fillIn(XmlFilePgeConfigBuilder.java:322)
    ... 12 more
java.io.IOException: Failed to build PgeConfig : Failed to parse value: SQL(FORMAT='$FileLocation/$Filename',SORT_BY='CAS.ProductReceivedTime'){SELECT FileLocation,Filename,CAS.ProductReceivedTime FROM RatLog}
    at org.apache.oodt.cas.pge.config.XmlFilePgeConfigBuilder.build(XmlFilePgeConfigBuilder.java:75)
    at org.apache.oodt.cas.pge.PGETaskInstance.createPgeConfig(PGETaskInstance.java:133)
    at org.apache.oodt.cas.pge.PGETaskInstance.initialize(PGETaskInstance.java:98)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:449)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
org.apache.oodt.cas.workflow.structs.exceptions.WorkflowTaskInstanceException: PGETask failed : Failed to instanciate PGETaskInstance : Failed to build PgeConfig : Failed to parse value: SQL(FORMAT='$FileLocation/$Filename',SORT_BY='CAS.ProductReceivedTime'){SELECT FileLocation,Filename,CAS.ProductReceivedTime FROM RatLog}
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:456)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.executeTaskLocally(IterativeWorkflowProcessorThread.java:574)
    at org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread.run(IterativeWorkflowProcessorThread.java:321)
    at EDU.oswego.cs.dl.util.concurrent.PooledExecutor$Worker.run(Unknown Source)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.InstantiationException: Failed to instanciate PGETaskInstance : Failed to build PgeConfig : Failed to parse value: SQL(FORMAT='$FileLocation/$Filename',SORT_BY='CAS.ProductReceivedTime'){SELECT FileLocation,Filename,CAS.ProductReceivedTime FROM RatLog}
    at org.apache.oodt.cas.pge.PGETaskInstance.initialize(PGETaskInstance.java:116)
    at org.apache.oodt.cas.pge.PGETaskInstance.run(PGETaskInstance.java:449)
    ... 4 more
Aug 31, 2015 9:30:17 PM org.apache.oodt.cas.workflow.engine.IterativeWorkflowProcessorThread executeTaskLocally
WARNING: Exception executing task: [RatAggregator] locally: Message: PGETask failed : Failed to instanciate PGETaskInstance : Failed to build PgeConfig : Failed to parse value: SQL(FORMAT='$FileLocation/$Filename',SORT_BY='CAS.ProductReceivedTime'){SELECT FileLocation,Filename,CAS.ProductReceivedTime FROM RatLog}
Aug 31, 2015 9:32:08 PM org.apache.oodt.cas.filemgr.catalog.LuceneCatalog paginateQuery
WARNING: Query: [q=] for Product Type: [urn:drat:RatLog] returned no results
```

**Comment by chrismattmann** [comment]
closed in #42

**Comment by chrismattmann** [comment]
thanks @lewismc !
